docker/clion-toolchain.dockerfile

```diff
@@ -0,0 +1,100 @@
+# syntax=docker/dockerfile:1.4
+# artifacts: false
+# platforms: linux/amd64
+# platforms_pr: linux/amd64
+# no-cache-filters: toolchain-base,toolchain
+ARG BASE=ubuntu
+ARG TAG=22.04
+FROM ${BASE}:${TAG} AS toolchain-base
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+FROM toolchain-base as toolchain
+
+ARG TARGETPLATFORM
+RUN echo "target_platform: ${TARGETPLATFORM}"
+
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+# install dependencies
+RUN <<_DEPS
+#!/bin/bash
+set -e
+apt-get update -y
+apt-get install -y --no-install-recommends \
+  build-essential \
+  cmake=3.22.* \
+  ca-certificates \
+  gcc=4:11.2.* \
+  g++=4:11.2.* \
+  gdb \
+  git \
+  libayatana-appindicator3-dev \
+  libavdevice-dev \
+  libboost-filesystem-dev=1.74.* \
+  libboost-locale-dev=1.74.* \
+  libboost-log-dev=1.74.* \
+  libboost-program-options-dev=1.74.* \
+  libcap-dev \
+  libcurl4-openssl-dev \
+  libdrm-dev \
+  libevdev-dev \
+  libminiupnpc-dev \
+  libnotify-dev \
+  libnuma-dev \
+  libopus-dev \
+  libpulse-dev \
+  libssl-dev \
+  libva-dev \
+  libvdpau-dev \
+  libwayland-dev \
+  libx11-dev \
+  libxcb-shm0-dev \
+  libxcb-xfixes0-dev \
+  libxcb1-dev \
+  libxfixes-dev \
+  libxrandr-dev \
+  libxtst-dev \
+  udev \
+  wget
+if [[ "${TARGETPLATFORM}" == 'linux/amd64' ]]; then
+  apt-get install -y --no-install-recommends \
+    libmfx-dev
+fi
+apt-get clean
+rm -rf /var/lib/apt/lists/*
+_DEPS
+
+#Install Node
+# hadolint ignore=SC1091
+RUN <<_INSTALL_NODE
+#!/bin/bash
+set -e
+node_version="20.9.0"
+wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/master/install.sh | bash
+source "$HOME/.nvm/nvm.sh"
+nvm install "$node_version"
+nvm use "$node_version"
+nvm alias default "$node_version"
+_INSTALL_NODE
+
+# install cuda
+WORKDIR /build/cuda
+# versions: https://developer.nvidia.com/cuda-toolkit-archive
+ENV CUDA_VERSION="11.8.0"
+ENV CUDA_BUILD="520.61.05"
+# hadolint ignore=SC3010
+RUN <<_INSTALL_CUDA
+#!/bin/bash
+set -e
+cuda_prefix="https://developer.download.nvidia.com/compute/cuda/"
+cuda_suffix=""
+if [[ "${TARGETPLATFORM}" == 'linux/arm64' ]]; then
+  cuda_suffix="_sbsa"
+fi
+url="${cuda_prefix}${CUDA_VERSION}/local_installers/cuda_${CUDA_VERSION}_${CUDA_BUILD}_linux${cuda_suffix}.run"
+echo "cuda url: ${url}"
+wget "$url" --progress=bar:force:noscroll -q --show-progress -O ./cuda.run
+chmod a+x ./cuda.run
+./cuda.run --silent --toolkit --toolkitpath=/usr/local/cuda --no-opengl-libs --no-man-page --no-drm
```

Comment:
@cgutman do you know what path we can use for the cuda toolkitpath, without having to specify `-DCMAKE_CUDA_COMPILER:PATH=/usr/local/cuda/bin/nvcc \`

The default doesn't work and neither does `/usr/local/cuda`
```txt
-- Looking for a CUDA compiler
-- Looking for a CUDA compiler - NOTFOUND
```

Or should I just sym link `/usr/local/cuda/bin/nvcc` to `/usr/bin/nvcc`?

Reply:
Nvidia's package probably doesn't symlink into `/usr/bin` since `/usr/local/bin` is the correct place for binaries that aren't managed via the distro's package manager. When CUDA is installed via the package manager (like Ubuntu's [nvidia-cuda-toolkit](https://packages.ubuntu.com/jammy/nvidia-cuda-toolkit) package), it actually [does include](https://packages.ubuntu.com/jammy/amd64/nvidia-cuda-toolkit/filelist) symlinks into /usr.

Reply:
Hmm... I'm really at a loss.

Using `toolkitpath=/usr/local` works, although I don't know what potential issues there are with that.

But using a symlink does not work (to either `/usr/bin/nvcc` or `/usr/local/bin/nvcc`)

```bash
./cuda.run --silent --toolkit --toolkitpath=/usr/local/cuda --no-opengl-libs --no-man-page --no-drm
rm ./cuda.run

# create symbolic link for nvcc
ln -sf /usr/local/cuda/bin/nvcc /usr/bin/nvcc
```

The symbolic link exists in the container, just not found by Cmake.

![image](https://github.com/LizardByte/Sunshine/assets/42013603/a7baeba4-ae80-438d-ad97-75ed7e92c6c5)

Reply:
> Using toolkitpath=/usr/local works, although I don't know what potential issues there are with that.

For a real system, you wouldn't want to pollute `/usr/local` (Nvidia used the `cuda` subfolder for a reason).

For a Docker image, probably nothing.